modules/layers/src/text-layer/text-layer.js

```diff
@@ -179,53 +179,40 @@ export default class TextLayer extends CompositeLayer {
     const {data, getText} = this.props;
     const {iconMapping} = this.state;
     const {startRow, endRow} = dataRange;
+    const {iterable, objectInfo} = createIterable(data, startRow, endRow);
 
     const transformedData = [];
 
-    const {iterable, objectInfo} = createIterable(data, startRow, endRow);
     for (const object of iterable) {
+      const transformCharacter = ({text, index, offsetLeft, offsetTop, size, rowSize, len}) => {
```

Comment:
Perf: This is a large amount of deconstructing and reconstructing small objects in a private function.